test/fakeIngestion/standalone/src/main/java/com/microsoft/applicationinsights/test/fakeingestion/MockedAppInsightsIngestionServer.java

```diff
@@ -104,6 +106,17 @@ public Envelope waitForItem(Predicate<Envelope> condition, int timeout, TimeUnit
         return waitForItems(condition, 1, timeout, timeUnit).get(0);
     }
 
+    public List<Envelope> waitForItems(final String type, final int numItems) throws Exception {
+        List<Envelope> items = waitForItems(new Predicate<Envelope>() {
+            @Override public boolean apply(Envelope input) {
+                return input.getData().getBaseType().equals(type);
+            }
+        }, numItems, 10, TimeUnit.SECONDS);
+        if (items.size() > numItems) {
+            throw new IllegalStateException("Received too many items of type " + type);
```

Comment:
@littleaj how about this?
```suggestion
            throw new AssertionError("Expecting " + numItems + " of type " + type + ", but received " + items.size());
```